array.c

```diff
@@ -960,58 +965,96 @@ rb_ary_s_try_convert(VALUE dummy, VALUE ary)
 
 /*
  *  call-seq:
- *     Array.new(size=0, default=nil)
- *     Array.new(array)
- *     Array.new(size) {|index| block }
+ *    Array.new -> new_empty_array
+ *    Array.new(array) -> new_array
+ *    Array.new(size) -> new_array
+ *    Array.new(size, default_value) -> new_array
+ *    Array.new(size) { |index| ... } -> new_array
+ *
+ *  Returns a new \Array.
+ *
+ *  Argument <tt>array</tt>, if given, must be an
+ *  {Array-convertible object}[doc/implicit_conversion_rdoc.html#label-Array-Convertible+Objects].
+ *
+ *  Argument <tt>size</tt>, if given must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
+ *
+ *  Argument <tt>default_value</tt> may be any object.
  *
- *  Returns a new array.
+ *  ---
```

Comment:
Nice way to separate creation methods!

Reply:
I like it, too.  I've used it, as here, to:
- Separate initial remarks about the args.
- Separate trailing exception cases.
Elsewhere I've used it to separate sections when there's more than one related example in a section.